File: easy/1572.py
class Solution(object):
    def diagonalSum(self, mat):
        """
        :type mat: List[List[int]]
        :rtype: int
        """
        sum = 0
        i = 0
        j = len(mat) - 1
        while i < len(mat):
            sum += mat[i][i]
            sum += mat[j][i]
            i += 1
            j -= 1
        if len(mat) % 2 != 0:
            sum -= mat[len(mat) / 2][len(mat) / 2]
        return sum
        

File: easy/1572.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum = 0;
        int j = 0;
        for (int i = 0; i < mat.length; i++){
            sum += mat[i][j];
            ++j;
        }
        int n = 0;
        for (int m = mat.length - 1; m >=0; m--){
            sum += mat[m][n];
            ++n;
        }
        if (mat.length % 2 != 0){
            sum -= mat[(mat.length / 2)][(mat.length / 2)];
        }
        return sum;

    }
}
